src/kimmdy/kmc.py

```diff
@@ -72,9 +72,11 @@ def rf_kmc(
     reaction_probability = []
 
     # 1. Calculate the probability for each reaction
+
     for recipe in recipe_collection.recipes:
         dt = [x[1] - x[0] for x in recipe.timespans]
-        reaction_probability.append(sum(np.multiply(dt, recipe.rates)))
+        t_interval = recipe.timespans[-1][-1] - recipe.timespans[0][0]
```

Comment:
This calculates the time between the first time this recipe becomes valid, and the last time it is valid. However, in between it might be invalid.
**If** the reaction probability should be scaled down by the time it should be ?
```
sum(dt)
```